tests/Unit/Position/PositionTest.php

```diff
@@ -0,0 +1,284 @@
+    {
+        $this->subject->setPosition(99);
+
+        $this->subject->setPosition(null);
+
+        self::assertNull($this->subject->getLineNumber());
+    }
+
+    /**
+     * @return array<non-empty-string, array{0: int<0, max>}>
+     */
+    public function provideColumnNumber(): array
+    {
+        return [
+            'column 0' => [0],
+            'column 14' => [14],
+            'column 39' => [39],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @param int<0, max> $columnNumber
+     *
+     * @dataProvider provideColumnNumber
+     */
+    public function setPositionOnVirginSetsColumnNumber(int $columnNumber): void
+    {
+        $this->subject->setPosition(1, $columnNumber);
+
+        self::assertSame($columnNumber, $this->subject->getColumnNumber());
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideColumnNumber
+     */
+    public function setPositionSetsNewColumnNumber(int $columnNumber): void
+    {
+        $this->subject->setPosition(1, 99);
+
+        $this->subject->setPosition(2, $columnNumber);
+
+        self::assertSame($columnNumber, $this->subject->getColumnNumber());
+    }
+
+    /**
+     * @test
+     */
+    public function setPositionWithoutColumnNumberClearsColumnNumber(): void
+    {
+        $this->subject->setPosition(1, 99);
+
+        $this->subject->setPosition(2);
+
+        self::assertNull($this->subject->getColumnNumber());
+    }
+
+    /**
+     * @test
+     */
+    public function setPositionWithNullForColumnNumberClearsColumnNumber(): void
+    {
+        $this->subject->setPosition(1, 99);
+
+        $this->subject->setPosition(2, null);
+
+        self::assertNull($this->subject->getColumnNumber());
+    }
+
+    /**
+     * @return array<non-empty-string, array{0: int<1, max>, 1: int<0, max>}>
+     */
+    public function provideLineAndColumnNumber(): array
+    {
+        return DataProviders::cross($this->provideLineNumber(), $this->provideColumnNumber());
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideLineAndColumnNumber
+     */
+    public function setPositionOnVirginSetsLineAndColumnNumber(int $lineNumber, int $columnNumber): void
+    {
+        $this->subject->setPosition($lineNumber, $columnNumber);
+
+        self::assertSame($lineNumber, $this->subject->getLineNumber());
+        self::assertSame($columnNumber, $this->subject->getColumnNumber());
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideLineAndColumnNumber
+     */
+    public function setPositionSetsNewLineAndColumnNumber(int $lineNumber, int $columnNumber): void
+    {
+        $this->subject->setPosition(98, 99);
+
+        $this->subject->setPosition($lineNumber, $columnNumber);
+
+        self::assertSame($lineNumber, $this->subject->getLineNumber());
+        self::assertSame($columnNumber, $this->subject->getColumnNumber());
+    }
+
+    // Testing deprecated functionality
```

Comment:
These should reside in `Tests/UnitDeprecated/` instead.

Reply:
Done.  I could not think of an easy way to share the data providers between the two `TestCase`s - once you extend `TestCase`, it's `final` - so have duplicated them.  The tests for deprecated functionality will be removed in time, so it's no big deal.